parser.go

```diff
@@ -220,19 +240,19 @@ type ParserConfig struct {
 //
 // See also: package fuzzy for existing emitters with fuzzy-logic that depends on the demo-type.
 type EventEmitter interface {
-	Register(parser *Parser, eventDispatcher func(event interface{}))
+	Register(parser *parser, eventDispatcher func(event interface{}))
```

Comment:
This should probably take the interface as arg

```suggestion
	Register(parser Parser, eventDispatcher func(event interface{}))
```